### PR TITLE
Fix #299: getNotDeletedSiteImagesAsCursor

### LINKED ISSUE #299: MediaStore.getNotDeleted* getters return empty results
We must investigate and write tests.

### PR DESCRIPTION
**Pull request by maxme**
Fix #299 - adds a few tests as well

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/MediaSqlUtils.java [hunk 1]
```diff
@@ -128,8 +128,8 @@
     public static SelectQuery<MediaModel> getMediaExcludingQuery(SiteModel site, String column, Object value) {
         return WellSql.select(MediaModel.class)
                 .where().beginGroup()
-                .equals(MediaModelTable.SITE_ID, site.getSiteId())
                 .not().equals(column, value)
+                .equals(MediaModelTable.SITE_ID, site.getSiteId())
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/MediaSqlUtils.java]: Is this change necessary? Can't we just bump the `wellsql` dependency to `1.1.1` and get the https://github.com/wordpress-mobile/wellsql/pull/6 fix?

Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/MediaSqlUtils.java]: Yes, but I was tire of waiting for the package sync

**Comment by tonyr59h** [comment]
:shipit:
